src/main/java/org/mockito/internal/util/eventbus/Subscriber.java

```diff
@@ -0,0 +1,78 @@
+package org.mockito.internal.util.eventbus;
+
+import static java.lang.System.identityHashCode;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import org.mockito.exceptions.base.MockitoException;
+
+public class Subscriber {
+
+    private final Object listenerInstance;
+    private final Method listenerMethod;
+    private final Class<?> eventType;
+
+    public Subscriber(Object listenerInstance, Method listenerMethod) {
+        this.listenerInstance = listenerInstance;
+        this.listenerMethod = listenerMethod;
+        this.eventType = listenerMethod.getParameterTypes()[0];
+    }
+
+    public final void tryHandleEvent(Object event) {
+        if (canHandle(event)) {
+            invokeListenerMethod(event);
+        }
+    }
+
+    protected void invokeListenerMethod(Object event) {
```

Comment:
This method feels very hacky. Especially since you are using reflection modify the accessibility. Would prefer a cleaner and nicer solution for this.